core/shared/src/main/scala/zio/prelude/ZValidation.scala

```diff
@@ -261,6 +271,21 @@ object ZValidation extends LowPriorityValidationImplicits {
 
   final case class Failure[+W, +E](log: Chunk[W], errors: NonEmptyChunk[E]) extends ZValidation[W, E, Nothing] {
     lazy val errorsUnordered: NonEmptyMultiSet[E] = NonEmptyMultiSet.fromIterable(errors.head, errors.tail)
+
+    lazy val message: String =
+      s"errors:\n  ${errors.map(_.toString).mkString("\n  ")}\nlog:\n  ${log.map(_.toString).mkString("\n  ")}"
+
+    def toException: Failure.Exception[W, E] = Failure.Exception(this)
+
+    def toExceptionWithCause(implicit ev: E <:< Throwable): Failure.Exception[W, E] = {
+      val exn = toException
+      errors.tail.foreach(exn.addSuppressed(_))
+      val _   = exn.initCause(errors.head)
+      exn
+    }
+  }
+  final object Failure {
+    final case class Exception[+W, +E](failure: Failure[W, E]) extends IllegalArgumentException(failure.message)
```

Comment:
Why not just wrap a `NonEmptyChunk[E]` here?

Reply:
Why `IllegalArgumentException`? Seems overly specific.

Reply:
I would look at the implementation of `FiberFailure` in ZIO. We could have something similar like `ValidationFailure` if we wanted.

Reply:
> Why not just wrap a `NonEmptyChunk[E]` here?

I thought it would be useful to have also the `log` available

> Why `IllegalArgumentException`? Seems overly specific.

The most typical use case for Validation is _validate_ input arguments, so `IllegalArgumentException` seemed like a natural choice. What do you think would be a better choice?

> `FiberFailure`

OK, I'll check it out :+1:

Reply:
That's fine. It could just be:

```scala
final case class ValidationFailure(failure: Validation.Failure[Any]) extends Throwable(null, null, true, false)
```

Reply:
I have reworked it as `ZValidationFailureException`